dracclient/resources/raid.py

```diff
@@ -98,15 +98,295 @@
+        """Creates RAIDIntegerAttribute object
+
+        :param name: name of the RAID attribute
+        :param instance_id: InstanceID of the RAID attribute
+        :param current_value: list containing the current value of the RAID
+		attribute
+        :param pending_value: pending value of the RAID attribute,
+                reflecting an unprocessed change (eg. config job not completed)
+        :param read_only: indicates whether this RAID attribute can be
+                changed
+        :param fqdd: Fully Qualified Device Description of the RAID
+                Attribute
+        :param lower_bound: minimum value for the RAID attribute
+        :param upper_bound: maximum value for the RAID attribute
+        """
+        super(RAIDIntegerAttribute, self).__init__(name, instance_id,
+                                                   current_value,
+                                                   pending_value,
+                                                   read_only, fqdd)
+        self.lower_bound = lower_bound
+        self.upper_bound = upper_bound
+
+
+    @classmethod
+    def parse(cls, raid_attr_xml):
+        """Parses XML and creates RAIDIntegerAttribute object"""
+
+        raid_attr = RAIDAttribute.parse(cls.namespace, raid_attr_xml)
+        lower_bound = utils.get_wsman_resource_attr(
+            raid_attr_xml, cls.namespace, 'LowerBound')
+        upper_bound = utils.get_wsman_resource_attr(
+            raid_attr_xml, cls.namespace, 'UpperBound')
+
+        if raid_attr.current_value:
+            raid_attr.current_value = int(raid_attr.current_value[0])
+        if raid_attr.pending_value:
+            raid_attr.pending_value = int(raid_attr.pending_value)
+
+        return cls(raid_attr.name, raid_attr.instance_id,
+                   raid_attr.current_value, raid_attr.pending_value,
+                   raid_attr.read_only, raid_attr.fqdd,
+                   int(lower_bound), int(upper_bound))
+
+    def validate(self, new_value):
+        """Validates new value"""
+
+        val = int(new_value)
+        if val < self.lower_bound or val > self.upper_bound:
+            msg = ('Attribute %(attr)s cannot be set to value %(val)d.'
+                   ' It must be between %(lower)d and %(upper)d.') % {
+                       'attr': self.name,
+                       'val': new_value,
+                       'lower': self.lower_bound,
+                       'upper': self.upper_bound}
+            return msg
+
+
 class RAIDManagement(object):
 
+    NAMESPACES = [(uris.DCIM_RAIDEnumeration, RAIDEnumerableAttribute),
+                  (uris.DCIM_RAIDString, RAIDStringAttribute),
+                  (uris.DCIM_RAIDInteger, RAIDIntegerAttribute)]
+
     def __init__(self, client):
         """Creates RAIDManagement object
 
         :param client: an instance of WSManClient
         """
         self.client = client
 
+    def list_raid_controller_settings(self, by_name=False):
```

Comment:
I don't think a "by_name" parameter here makes sense because there are several attributes that are per virtual disk, which means there could be multiple attributes with the same name, but different instance IDs for a single RAID controller.  Let's remove this parameter to keep users from shooting themselves in the foot.